pandas/tests/indexes/test_base.py

```diff
@@ -2821,3 +2821,11 @@ def test_shape_of_invalid_index():
 
     idx = pd.Index([0, 1, 2, 3])
     assert idx[:, None].shape == (4, 1)
+
+
+def test_index_construction_respects_dtype():
+    index_list = [7606741985629028552, 17876870360202815256]
```

Comment:
I think it might be a little better to make this `[1, 2**63 + 1]` so we're testing at the exact boundary of the issue (no precision loss for 2**63).